### PR TITLE
Thulasizwe/en/1877: button and button group styling

### PR DESCRIPTION
**Pull request by czwe-01**
Addressing [issue](https://github.com/shesha-io/shesha-framework/issues/2038). Recent updates to the button and button group components have caused several issues affecting the layout and functionality of existing configurations.

- [ ]  The button layout of existing button configurations has been altered
- [ ]  Changes to border type, border width, and background color reflect immediately
- [ ] Colorpicker has clear button
- [ ] Configurator reflects the correct default button width
- [ ] Button group size applies to all buttons within the group
- [ ] Button type applied to buttons in a group.
- [ ] Hidden property re-added

### CODE REVIEW

File: shesha-reactjs/src/components/buttonGroupConfigurator/buttonGroupItem.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ export const ButtonGroupItem: FC<IButtonGroupItemProps> = ({ item, actualModelCo
   const { styles } = useStyles();
   const actualItem = useMemo(() => getActualModel({ ...item, actionConfiguration }, actualModelContext)
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    , [item.label, item.icon, item.tooltip, item.name, actualModelContext]);
+    , [{ ...item }, actualModelContext]);
```
Comment (AlexStepantsov) [inline shesha-reactjs/src/components/buttonGroupConfigurator/buttonGroupItem.tsx]: You must not do that, your implementation just always creates a new object and the recalculation will be performed at each rendering regardless of changes in properties

File: shesha-reactjs/src/components/buttonGroupConfigurator/properties.tsx
Comment (AlexStepantsov) [inline shesha-reactjs/src/components/buttonGroupConfigurator/properties.tsx:1]: Why are these changes here?

File: shesha-reactjs/src/providers/form/models.ts [hunk 2]
```diff
@@ -189,6 +190,8 @@ export interface IConfigurableFormComponent
   noDataSecondaryText?: string;
 
   permissions?: string[];
+
+  allowClear?: boolean;
```
Comment (AlexStepantsov) [inline shesha-reactjs/src/providers/form/models.ts]: Why is this property here? It is clearly specific to some component (I assume ColorPicker). What exactly in ALL components should be allowed to be cleared?

File: shesha-reactjs/src/components/buttonGroupConfigurator/buttonGroupItem.tsx (hunk 1 above)
Comment (czwe-01) [reply shesha-reactjs/src/components/buttonGroupConfigurator/buttonGroupItem.tsx]: When adding item as a dependency as it is, changes did not reflect the applied styles immediately since , till I started listing the style properties, which seemed a bit excessive because there were many properties to list, but now that you explain it, I see that my approach is not ideal and can affect performance.

File: shesha-reactjs/src/providers/form/models.ts (hunk 2 above)
Comment (czwe-01) [reply shesha-reactjs/src/providers/form/models.ts]: I noticed that it was supported in a couple of input components e.g. Input, datePicker, colorPicker, autocomplete, and Select, so I thought moving it up was a good idea.  Now I see it's not because very few components support it.

**Review by AlexStepantsov — APPROVED** [review]
